pyrf/gui/controller.py

```diff
@@ -586,3 +601,83 @@ def user_xrange_changed(self, start, stop):
 
     def applying_user_xrange(self):
         return self._applying_user_xrange
+
+    def save_settings(self, dir):
+        cfgfile = open(dir,'w')
+        config = ConfigParser.SafeConfigParser()
+        state = self._state.to_json_object()
+
+        config.add_section('device_options')
+        for s in state:
+            if 'dict' in str(type(state[s])):
```

Comment:
Hmm. shouldn't usually need type checking in this kind of code.

The better way to check types (when it's needed, which I still suspect it isnt') is `isinstance(s, dict)`